ARCHITECTURE.md

```diff
@@ -0,0 +1,108 @@
+- [`./scripts/`](scripts/): Local utility scripts, not used by the production application
+
+## Major website areas
+
+### Homepage
+
+The homepage ([url rule](webapp/app.py#L637)) has no server-side dynamic functionality, so is instead served directly by the template finder view from `templates/index.html`.
+
+Since ubuntu.com is a high-traffic site, and the homepage its most popular page, it is important that the initial response for this page remains as fast as possible, so it should *not* be making database or API calls directly if it can be avoided.
+
+#### Homepage takeovers and engage pages
+
+The main central area of the homepage contains a "takeover" - a large strip advertising a Canonical product or service. These takeovers often then link to "engage pages" - a page about a specific piece of marketing content like a whitepaper.
+
+The list of takeovers is maintained by the Marketing team, and when the homepage loads, a random takeover will be chosen and then on reloads it will cycle through the list of takeovers, so you can see all takeovers by refreshing a few times.
+
+The takeovers are loaded client-side by calling https://ubuntu.com/takeovers.json ([url rule](webapp/app.py#L602), [view function](webapp/app.py#L568-L578)). You can also see all available takeovers listed at https://ubuntu.com/takeovers ([url rule](webapp/app.py#L603), [view function](webapp/app.py#L581-L598)), and you can see available engage pages at https://ubuntu.com/engage.
+
+The takeover content is maintained by Marketing in [the takeovers category in Discourse](https://discourse.ubuntu.com/c/design/takeovers/106), and the /takeovers.json view pulls the content from here through the Discourse API using [our Discourse module](https://github.com/canonical/canonicalwebteam.discourse) ([instantiated here](webapp/app.py#L510-L515)). Similarly, the engage pages are maintained in [a Discourse category](https://discourse.ubuntu.com/c/design/engage-pages/51), which populates https://ubuntu.com/engage.
+
+Here's [a more complete guide to the engage pages and takeovers system](https://discourse.canonical.com/t/engage-pages-and-takeovers-v2/358).
+
+### Blog
+
+The blog pages under https://ubuntu.com/blog make use of [the blog module](https://github.com/canonical/canonicalwebteam.blog) ([instantiated here](webapp/app.py#L437-L442)) to pull in content from the `admin.insights.ubuntu.com` Wordpress installation through the Wordpress API. This content is managed through [the Wordpress admin area](https://admin.insights.ubuntu.com/admin), which is only accessible while on [the company VPN](https://wiki.canonical.com/InformationInfrastructure/IS/HowTo/CompanyOpenVPN).
+
+### Documentation
+
+There are a large number of documentation areas on ubuntu.com:
+
+- [ubuntu.com/servers/docs](http://ubuntu.com/servers/docs) ([code](webapp/app.py#L642-L651))
```

Comment:
Might be a nitpick-y/silly question, but is the aim to make this a complete list? If not then it might be a good idea to flag that it isn't for clarity's sake. "There are a large number of documentation areas on ubuntu.com. These include but are not limited to:" or something

Reply:
I think it's complete at the time of writing. I could say it's complete at the time of writing.

Reply:
Yup, that sounds good. I think it's missing k8s docs though https://ubuntu.com/kubernetes/docs